utils/analysis/factory.go

```diff
@@ -219,6 +219,12 @@ func ValidateMetric(metric v1alpha1.Metric) error {
 	if metric.Provider.CloudWatch != nil {
 		numProviders++
 	}
+	if metric.Provider.Graphite != nil {
+		numProviders++
+	}
```

Comment:
Quick question, why does this introduce graphite?

Reply:
Same as above - I fixed same graphite issues along the way.

Reply:
Do you think I should drop the graphite fixup? I just noticed it was missing here.

Reply:
Ahh good question. I'd say contribute it and let Argo team decided if they want to split it out.